single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/layout/LegBusSet.java

```diff
@@ -232,15 +231,59 @@ public static void pushLBS(List<LegBusSet> legBusSets, LegBusSet legBusSet) {
         legBusSets.add(legBusSet);
     }
 
-    private static void pushNonUnilegInternCell(List<LegBusSet> legBusSets, Map<BusNode, Integer> nodeToNb, InternCell internCell) {
+    private static void pushInternCell(List<LegBusSet> legBusSets, Map<BusNode, Integer> nodeToNb, InternCell internCell) {
+        List<LegBusSet> attachedLegBusSets = new ArrayList<>();
         for (LegBusSet lbs : legBusSets) {
-            if (lbs.contains(internCell.getBusNodes())) {
-                lbs.addInternCell(internCell, Side.UNDEFINED);
-                internCell.setShape(InternCell.Shape.VERTICAL);
-                return;
+            boolean attachedToLbs = internCell.getBusNodes().stream().anyMatch(lbs.busNodeSet::contains);
+            if (attachedToLbs) {
+                attachedLegBusSets.add(lbs);
+                if (lbs.busNodeSet.containsAll(internCell.getBusNodes())) {
+                    lbs.addInternCell(internCell, Side.UNDEFINED);
+                    if (internCell.getShape() == InternCell.Shape.MAYBE_ONE_LEG) {
+                        internCell.setShape(InternCell.Shape.ONE_LEG);
+                    } else {
+                        internCell.setShape(InternCell.Shape.VERTICAL);
+                    }
+                    return;
+                }
             }
         }
-        pushLBS(legBusSets, new LegBusSet(nodeToNb, internCell, Side.LEFT));
-        pushLBS(legBusSets, new LegBusSet(nodeToNb, internCell, Side.RIGHT));
+        if (internCell.getShape() == InternCell.Shape.MAYBE_ONE_LEG) {
+            replaceByMultilegOrSetOneLeg(internCell, attachedLegBusSets);
+        }
+        if (internCell.getShape() != InternCell.Shape.ONE_LEG) {
+            pushLBS(legBusSets, new LegBusSet(nodeToNb, internCell, Side.LEFT));
+            pushLBS(legBusSets, new LegBusSet(nodeToNb, internCell, Side.RIGHT));
+        } else {
+            pushLBS(legBusSets, new LegBusSet(nodeToNb, internCell, Side.UNDEFINED));
+        }
+    }
+
+    private static void replaceByMultilegOrSetOneLeg(InternCell internCell, List<LegBusSet> attachedLegBusSets) {
+        // the one leg intern cell cannot be absorbed by any existing legBusSet
+        // we consider that a one leg intern cell should not force the corresponding busNodes
+        // to be parallel, hence we try to replace that one leg by a multileg
+        LegBlock oneLeg = internCell.getSideToLeg(Side.UNDEFINED);
+        if (oneLeg instanceof LegParallelBlock legParallelBlock) {
+            List<LegPrimaryBlock> subBlocks = legParallelBlock.getSubBlocks();
+            if (subBlocks.size() == 2) {
+                internCell.replaceOneLegByMultiLeg(subBlocks.get(0), subBlocks.get(1));
+            } else {
```

Comment:
This else statement is never used (neither in unit tests nor in other tests I ran with French substations).
Shall we remove it?